### PR TITLE
Meilisearch double quote on "match" query

### PR DESCRIPTION
**Pull request by 6543**
make `nonFuzzyWorkaround` unessesary

cc @Kerollmops

### CODE REVIEW

**Comment by 6543** [comment]
> Thank you for the clear explanation. So, exposing the typoTolerance.enabled setting on the search route could fix the issue?

PS: yes t hat would be even better ...

**Comment by Kerollmops** [comment]
> Would that work?

Yeah, that's the idea. The only downside I see with this "double-quote everything" solution is that you cannot parse the same way as Meilisearch, and that's why the "expose the `typoTolerance.enabled` setting in the search route" is better in my opinion.

I just talked about this to our Product manager and they will prioritize this.

However, we don't refer to typo tolerance as fuzziness so I was a bit confused 😄

**Comment by 6543** [comment]
@Kerollmops that's awesome :heart: 

I hope to get the code-search ready for meilisearch too and at that point we should have a really good integration :rocket:

**Comment by 6543** [comment]
well with this patch if I search for `Tabellenspalte` in the code:

```
... in der `Tabellenspalte "D"` ist ... 
```

it wont find it anymore but I think that's ok as #29701 will let the user decide if he want it fuzzy (typoTolerance) or not

thanks in advance to @**Kerollmops** for  pointing out how to tell meilisearch to do so

**Comment by Kerollmops** [comment]
Thank you for the PR @6543. We will take care of it later this or next week 🙂

File: modules/indexer/issues/meilisearch/meilisearch.go [hunk 1]
```diff
@@ -247,73 +253,31 @@ func parseSortBy(sortBy internal.SortBy) string {
...
-// nonFuzzyWorkaround is needed as meilisearch does not have an exact search
-// and you can only change "typo tolerance" per index. So we have to post-filter the results
-// https://www.meilisearch.com/docs/learn/configuration/typo_tolerance#configuring-typo-tolerance
-// TODO: remove once https://github.com/orgs/meilisearch/discussions/377 is addressed
-func nonFuzzyWorkaround(searchRes *meilisearch.SearchResponse, keyword string, isFuzzy bool) ([]internal.Match, error) {
+func doubleQuoteKeyword(k string) string {
+	kp := strings.Split(k, " ")
+	parts := 0
+	for i := range kp {
+		if kp[i] != "" {
```
Comment (lunny) [inline modules/indexer/issues/meilisearch/meilisearch.go]: This is impossible?

**Comment by lunny** [comment]
Can you have a ref doc link for the quote behaviours?

**Comment by Kerollmops** [comment]
Here it is: https://www.meilisearch.com/docs/reference/api/search#phrase-search

File: modules/indexer/issues/meilisearch/meilisearch.go (hunk 1 above)
Comment (6543) [reply modules/indexer/issues/meilisearch/meilisearch.go]: @lunny yes it is:

https://play.golang.com/p/8R2e_x-eTCC
-> `[]string{"a", "", "b", "c"}`

**Review by silverwind — APPROVED** [review]

File: modules/indexer/issues/meilisearch/meilisearch.go [hunk 2]
```diff
@@ -247,73 +255,31 @@ func parseSortBy(sortBy internal.SortBy) string {
...
+			kp[parts] = fmt.Sprintf(`"%s"`, kp[i])
```
Comment (lunny) [inline modules/indexer/issues/meilisearch/meilisearch.go]: Maybe we should check whether it already has double quotes.

Comment (lunny) [reply modules/indexer/issues/meilisearch/meilisearch.go]: The last one from me, otherwise LGTM.

Comment (6543) [reply modules/indexer/issues/meilisearch/meilisearch.go]: done -> 6b9f4ebe8bc8a21805fc20513b63a85225801ed5

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
#29846

**Pull request merged**